### PR TITLE
Add kubebuilder as a way to use custom resources

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes-sigs/kubebuilder/issues/481

The definition is taken from here: https://book.kubebuilder.io/getting_started/what_is_kubebuilder.html.

/assign @droot @sttts

### CODE REVIEW

File: content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md [hunk 1]
```diff
@@ -236,6 +236,7 @@ When you add a custom resource, you can access it using:
 - The kubernetes dynamic client.
 - A REST client that you write.
 - A client generated using [Kubernetes client generation tools](https://github.com/kubernetes/code-generator) (generating one is an advanced undertaking, but some projects may provide a client along with the CRD or AA).
+- The [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder) SDK, which builds on top of the canonical techniques used to build the core Kubernetes APIs to provide simple abstractions and reduce boilerplate and toil.
```
Comment (sttts) [inline content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: then also add operator sdk

Comment (droot) [reply content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: This is a good start @nikhita , but I do think we need a section other than `Accessing a custom resource` which presents `Kubebuilder` as an SDK to extend Kubernetes APIs.

Comment (droot) [reply content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md]: @sttts @nikhita can the addition of operator-sdk be done as a follow up PR ?

**Comment by mistyhacks** [comment]
Please reopen this PR when you get a chance to address pending feedback. Thanks!

**Pull request closed without merging**